python3Packages.dropbox: Clean up package and enable more tests

SFrijters opened the pull request:
###### Description of changes

Followup to #170261 .
Clean up dependencies, enable the unittests at least.

~Open issue (tips welcome):~

```
$ ls result/lib/python3.9/site-packages/
dropbox  dropbox-0.0.0.dist-info
```

~Any idea why this would get a 0.0.0 version? This (probably) causes `maestral` to fail to find a matching version.~

@peterhoeg

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

SFrijters committed:
python3Packages.dropbox: Clean up package and enable more tests

SFrijters commented:
Found the issue I think; see 2nd commit.

thiagokokada commented on pkgs/development/python-modules/dropbox/default.nix:
```diff
@@ -39,6 +39,8 @@ buildPythonPackage rec {
   postPatch = ''
     substituteInPlace setup.py \
       --replace "'pytest-runner == 5.2.0'," ""
+    substituteInPlace dropbox/dropbox_client.py \
+      --replace "__version__ = '0.0.0'" "__version__ = '${version}'"
```
Should we report this error upstream?

Also, it is safe to assume this was a mistake right? Probably add a comment here to remove it on next release (or at least check if necessary).

SFrijters replied:
Yeah I just did that at https://github.com/dropbox/dropbox-sdk-python/issues/425 .
Depending on the answer I get I will add a comment to remove or keep (this substitution should be safe to have around as a fallback).

thiagokokada replied:
I would still leave a comment here at least pointing to the issue, just for context in the future.

SFrijters replied:
Package was fixed upstream but the same tag/version was re-used so I updated the hash instead to get the latest. This fixes `maestral` and friends (again).

Mindavi commented:
Diff LGTM, some testing would be nice though.

SFrijters commented:
Result of `nixpkgs-review pr 170289` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>git-annex-remote-dbx</li>
    <li>maestral (python39Packages.maestral)</li>
    <li>maestral-gui</li>
    <li>python310Packages.dropbox</li>
    <li>python310Packages.maestral</li>
    <li>python39Packages.dropbox</li>
  </ul>
</details>

SFrijters committed:
python3Packages.dropbox: Update hash for re-release of 11.30.0

The version of the package was not set properly and defaulted to 0.0.0 .
The package was re-released by upstream (with the same version number)
after it was reported.

This package will now have a correct version number in the dist-info when built,
fixing packages that require it to be within certain version bounds.

thiagokokada commented:
Result of `nixpkgs-review pr 170289` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>git-annex-remote-dbx</li>
    <li>maestral (python39Packages.maestral)</li>
    <li>maestral-gui</li>
    <li>python310Packages.dropbox</li>
    <li>python310Packages.maestral</li>
    <li>python39Packages.dropbox</li>
  </ul>
</details>

thiagokokada reviewed (commented):
Thanks and sorry for the issues.
